lib/measured/conversion_table.rb

```diff
@@ -0,0 +1,67 @@
+class Measured::ConversionTable
+
+  def initialize(units)
+    @units = units
+  end
+
+  def to_h
+    table = {}
+
+    @units.map{|u| u.name}.each do |to_unit|
+      to_table = {to_unit => BigDecimal("1")}
+
+      table.each do |from_unit, from_table|
+        to_table[from_unit] = find_conversion(to: from_unit, from: to_unit)
+        from_table[to_unit] = find_conversion(to: to_unit, from: from_unit)
+      end
+
+      table[to_unit] = to_table
+    end
+
+    table
+  end
+
+  private
+
+  def find_conversion(to:, from:)
+    conversion = find_direct_conversion(to: to, from: from) || find_tree_traversal_conversion(to: to, from: from)
+
+    raise Measured::UnitError, "Cannot find conversion path from #{ from } to #{ to }." unless conversion
+
+    conversion
+  end
+
+  def find_direct_conversion(to:, from:)
+    @units.each do |unit|
+      return unit.conversion_amount if unit.name == from && unit.conversion_unit == to
+    end
+
+    @units.each do |unit|
+      return BigDecimal(1) / unit.conversion_amount if unit.name == to && unit.conversion_unit == from
```

Comment:
Basically:
* Try to find a direct X -> Y conversion
* If not found, try to find the inverse Y -> X and convert it with 1/
* If not found do a recursive tree traversal to evaluate all possible paths to find an indirect conversion path